app/src/main/kotlin/at/bitfire/davdroid/ui/account/AccountSettingsModel.kt

```diff
@@ -120,33 +132,39 @@ class AccountSettingsModel @AssistedInject constructor(
     fun updateTasksSyncInterval(syncInterval: Long) {
         CoroutineScope(Dispatchers.Default).launch {
             tasksProvider?.authority?.let { tasksAuthority ->
+                val accountSettings = accountSettingsFactory.create(account)
                 accountSettings.setSyncInterval(tasksAuthority, syncInterval)
                 reload()
             }
         }
     }
 
-    fun updateSyncWifiOnly(wifiOnly: Boolean) {
+    fun updateSyncWifiOnly(wifiOnly: Boolean) = CoroutineScope(Dispatchers.Default).launch {
+        val accountSettings = accountSettingsFactory.create(account)
```

Comment:
Idea: instead of having this line in every function, we could make `accountSettings` lazy. Then we would just have to make sure that it's not evaluated in the main thread (+ KDoc that describes that).

What do you think?

Reply:
Sounds good. It was irritating me quite a lot but I forgot we could use lazy for that!